sim/warrior/protection/protection.go

```diff
@@ -37,35 +40,88 @@ func NewProtectionWarrior(character *core.Character, options *proto.Player) *Pro
 		Options: protOptions,
 	}
 
-	// rbo := core.RageBarOptions{
-	// 	StartingRage:   protOptions.ClassOptions.StartingRage,
-	// 	RageMultiplier: core.TernaryFloat64(war.Talents.EndlessRage, 1.25, 1),
-	// }
-	// if mh := war.GetMHWeapon(); mh != nil {
-	// 	rbo.MHSwingSpeed = mh.SwingSpeed
-	// }
-	// if oh := war.GetOHWeapon(); oh != nil {
-	// 	rbo.OHSwingSpeed = oh.SwingSpeed
-	// }
+	rbo := core.RageBarOptions{
+		StartingRage: protOptions.ClassOptions.StartingRage,
 
-	// war.EnableRageBar(rbo)
-	// war.EnableAutoAttacks(war, core.AutoAttackOptions{
-	// 	MainHand:       war.WeaponFromMainHand(war.DefaultMeleeCritMultiplier()),
-	// 	OffHand:        war.WeaponFromOffHand(war.DefaultMeleeCritMultiplier()),
-	// 	AutoSwingMelee: true,
-	// 	ReplaceMHSwing: war.TryHSOrCleave,
-	// })
-
-	// healingModel := options.HealingModel
-	// if healingModel != nil {
-	// 	if healingModel.InspirationUptime > 0.0 {
-	// 		core.ApplyInspiration(war.GetCharacter(), healingModel.InspirationUptime)
-	// 	}
-	// }
+		OnHitDealtRageGain: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult, rage float64) float64 {
+			if result.Target != nil && result.Target.CurrentTarget != &war.Unit {
+				return rage * 1.5 // Sentinel: Generate 50% addl rage from attacking targets not attacking the warrior
+			}
+
+			return rage
+		},
+	}
+	if mh := war.GetMHWeapon(); mh != nil {
+		rbo.MHSwingSpeed = mh.SwingSpeed
+	}
+
+	war.EnableRageBar(rbo)
+	war.EnableAutoAttacks(war, core.AutoAttackOptions{
+		MainHand:       war.WeaponFromMainHand(war.DefaultMeleeCritMultiplier()),
+		AutoSwingMelee: true,
+	})
+
+	war.RegisterSpecializationEffects()
```

Comment:
This should be called from the specs Initialize() as you dont have all stats created during the spec struct creation.